api/apps/api/test/project/delete-project-feature.e2e-spec.ts

```diff
@@ -0,0 +1,267 @@
+import { UsersProjectsApiEntity } from '@marxan-api/modules/access-control/projects-acl/entity/users-projects.api.entity';
+import { GivenUserExists } from '../steps/given-user-exists';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+afterEach(async () => {
+  await fixtures?.cleanup();
+});
+
+describe('Project - Delete Feature', () => {
+  test('should return NotFound error when project does not exist', async () => {
+    const nonExistentProjectId = v4();
+    const result = await fixtures.WhenDeletingFeatureForProject(
+      nonExistentProjectId,
+      v4(),
+    );
+    await fixtures.ThenProjectOrFeatureWasNotFound(
+      result,
+      `Project with id ${nonExistentProjectId} not found`,
+    );
+  });
+
+  test('should return NotFound error when feature does not exist', async () => {
+    const featureId = v4();
+    const projectId = fixtures.projectId;
+    const result = await fixtures.WhenDeletingFeatureForProject(
+      projectId,
+      featureId,
+    );
+    await fixtures.ThenProjectOrFeatureWasNotFound(
+      result,
+      `Feature with id ${featureId}, for project with id ${projectId}, not found`,
+    );
+  });
+
+  test('should not permit updating a given feature when the feature is associated to a different project', async () => {
+    const projectId = fixtures.projectId;
+    const anotherProjectId = fixtures.anotherProjectId;
+    const featureId = await fixtures.GivenBaseFeature(
+      'differentProject',
+      anotherProjectId,
+    );
+    const result = await fixtures.WhenDeletingFeatureForProject(
+      projectId,
+      featureId,
+    );
+    await fixtures.ThenDeleteWasForbidden(
+      result,
+      featureId,
+      `Feature with id ${featureId}, for project with id ${projectId}, cannot be deleted`,
+    );
+  });
+
+  test('should not permit updating a given feature, when it is not a custom feature', async () => {
+    const projectId = fixtures.projectId;
+    const featureId = await fixtures.GivenBaseFeature('customFeature');
+    const result = await fixtures.WhenDeletingFeatureForProject(
+      projectId,
+      featureId,
+    );
+    await fixtures.ThenDeleteWasForbidden(
+      result,
+      featureId,
+      `Feature with id ${featureId}, for project with id ${projectId}, cannot be deleted`,
+    );
+  });
+
+  test('should not permit updating a given feature, when it is still linked to a scenario', async () => {
+    const viewOnlyProjectId = fixtures.anotherProjectId;
+    const featureId = await fixtures.GivenBaseFeature(
+      'linkedFeature',
+      viewOnlyProjectId,
+    );
+    const result = await fixtures.WhenDeletingFeatureForProject(
+      viewOnlyProjectId,
+      featureId,
+    );
+    await fixtures.ThenDeleteWasForbidden(
+      result,
+      featureId,
+      `Feature with id ${featureId}, for project with id ${viewOnlyProjectId}, cannot be deleted`,
+    );
+  });
```

Comment:
ok, still a quick glance but I think the reason why tests don't catch the issue that I flagged in my review about linking of features to scenarios not being correctly checked is that here we're testing again that a feature cannot be deleted if the user only has read-only access to a project (the following test, basically) - at least that's what the naming of fixtures used here suggests - I haven't followed the full flow